SQL/New_patches/2018-05-28_publication_schema.sql

```diff
@@ -0,0 +1,169 @@
+SET FOREIGN_KEY_CHECKS=0;
+-- Publication Status
+DROP TABLE IF EXISTS publication_status;
+CREATE TABLE `publication_status` (
+  `PublicationStatusID` int(2) unsigned NOT NULL AUTO_INCREMENT,
+  `Label` varchar(255) NOT NULL,
+  PRIMARY KEY(`PublicationStatusID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+INSERT INTO publication_status (`Label`) VALUES ('Pending');
+INSERT INTO publication_status (`Label`) VALUES ('Approved');
+INSERT INTO publication_status (`Label`) VALUES ('Rejected');
+
+-- Main table
+DROP TABLE IF EXISTS publication;
+CREATE TABLE `publication` (
+    `PublicationID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+    `PublicationStatusID` int(2) unsigned NOT NULL default 1,
+    `LeadInvestigatorID` int(10) unsigned NOT NULL,
+    `UserID` int(10) unsigned NOT NULL,
+    `RatedBy` int(10) unsigned,
+    `DateProposed` date NOT NULL,
+    `DateRated` date default NULL,
+    `Title` varchar(255) NOT NULL,
+    `RejectedReason` varchar(255) default NULL,
+    `Description` text NOT NULL,
+    PRIMARY KEY(`PublicationID`),
+    CONSTRAINT `FK_publication_1` FOREIGN KEY(`UserID`) REFERENCES `users` (`ID`),
+    CONSTRAINT `FK_publication_2` FOREIGN KEY(`RatedBy`) REFERENCES `users` (`ID`),
+    CONSTRAINT `FK_publication_3` FOREIGN KEY(`PublicationStatusID`) REFERENCES `publication_status` (`PublicationStatusID`),
+    CONSTRAINT `FK_publication_4` FOREIGN KEY(`LeadInvestigatorID`) REFERENCES `publication_collaborator` (`PublicationCollaboratorID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+-- Separate table for Keywords
+DROP TABLE IF EXISTS publication_keyword;
+CREATE TABLE `publication_keyword` (
+  `PublicationKeywordID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `Label` varchar(255) NOT NULL,
+  PRIMARY KEY(`PublicationKeywordID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+-- Publication - Keyword relational table
+DROP TABLE IF EXISTS publication_keyword_rel;
+CREATE TABLE `publication_keyword_rel` (
+  `PublicationID` int(10) unsigned NOT NULL,
+  `PublicationKeywordID` int(10) unsigned NOT NULL,
+  PRIMARY KEY(PublicationID, PublicationKeywordID),
+  CONSTRAINT `FK_publication_keyword_1` FOREIGN KEY(`PublicationID`) REFERENCES `publication` (`PublicationID`),
+  CONSTRAINT `FK_publication_keyword_2` FOREIGN KEY(`PublicationKeywordID`) REFERENCES `publication_keyword` (`PublicationKeywordID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
+
+DROP TABLE IF EXISTS publication_collaborator;
+CREATE TABLE `publication_collaborator` (
+  `PublicationCollaboratorID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `Name` varchar(255) NOT NULL,
+  `Email` varchar(255),
+  PRIMARY KEY(`PublicationCollaboratorID`),
+  CONSTRAINT `UK_publication_collaborator_Email` UNIQUE (`Email`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8mb4';
+
```

Comment:
@davidblader When I tried running above sql patches on ibis-dev database, it gave me error saying;
"Error Code: 1071. Specified key was too long; max key length is 767 bytes." 
Also, the publication_collaborator table didn't get created. Changed the varchar limit to 191 and it fixed the issue.
[Ref:](https://stackoverflow.com/questions/1814532/1071-specified-key-was-too-long-max-key-length-is-767-bytes)

There was no problem when I tried it on my database. Anyway want to let you know.